packages/flame/lib/src/components/component.dart

```diff
@@ -243,6 +244,28 @@ class Component {
     }
   }
 
+  Map<Type, InheritedComponent>? _inheritedComponents;
+
+  T? dependOnInheritedComponentOfExactType<T extends InheritedComponent>() {
+    return _inheritedComponents?[T] as T?;
+  }
+
+  void _updateInheritance() {
+    final incomingComponents = _parent?._inheritedComponents;
+    if (incomingComponents == null) {
+      _inheritedComponents = null;
+    } else {
+      _inheritedComponents =
+          HashMap<Type, InheritedComponent>.from(incomingComponents);
+    }
+
+    final parent = this.parent;
+    if (parent is InheritedComponent) {
+      _inheritedComponents ??= HashMap<Type, InheritedComponent>();
```

Comment:
I think we shall make sure to set _inheritedComponents to null when the component is removed. Wdyt?